include/staggered_kd_build_xinv.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <gauge_field.h>
+
+namespace quda {
+
+  /**
+     @brief Build the Kahler-Dirac inverse block for KD operators.
+     @param[out] out Xinv resulting Kahler-Dirac inverse (assumed allocated)
+     @param[in] in gauge original fine gauge field
+     @param[in] in mass the mass of the original staggered operator w/out factor of 2 convention
+  */
+  void BuildStaggeredKahlerDiracInverse(GaugeField &Xinv, const cudaGaugeField &gauge, const double mass);
+
+  /**
+     @brief Allocate and build the Kahler-Dirac inverse block for KD operators
+     @param[in] in gauge original fine gauge field
+     @param[in] in mass the mass of the original staggered operator w/out factor of 2 convention
+     @param[in] in precision of Xinv field
+     @return constructed Xinv, which needs to be deleted manually
+  */
+  cudaGaugeField* AllocateAndBuildStaggeredKahlerDiracInverse(const cudaGaugeField &gauge, const double mass, const QudaPrecision override_prec);
```

Comment:
AS noted elsewhere, why `cudaGaugeField` and not `GaugeField`?

Reply:
Fixed return type. At this point the input fine gauge field will always be on the GPU; once that changes there will be a lot of places were we need to figure out the same logic of how `gauge` needs to be transformed.